tests/nativeapp/codegen/templating/test_templates_processor.py

```diff
@@ -213,3 +217,24 @@ def test_file_with_undefined_variable():
         assert "does not contain a valid template" in str(e.value)
         assert bundle_result.output_files[0].is_symlink()
         assert bundle_result.output_files[0].read_text() == file_contents[0]
+
+
+@mock.patch(CLI_GLOBAL_TEMPLATE_CONTEXT, {})
+@mock.patch(f"{TEMPLATE_PROCESSOR}.cc.warning")
+def test_expand_templates_in_file_unicode_decode_error(mock_cc_warning):
+    file_name = ["test_file.txt"]
+    file_contents = ["This is a test file"]
+    with TemporaryDirectory() as tmp_dir:
+        bundle_result = bundle_files(tmp_dir, file_name, file_contents)
+        templates_processor = TemplatesProcessor(bundle_ctx=bundle_result.bundle_ctx)
+        with mock.patch(
+            f"{ARTIFACT_PROCESSOR}.ProjectFileContextManager.__enter__",
```

Comment:
why not just mock `edit_file`? That's the public contract, whereas this is an implementation detail.